src/extension.ts

```diff
@@ -511,7 +511,7 @@ export function activate(context: vscode.ExtensionContext) {
                     cmdApp = espota; // Have binary EXE on Mac/Windows
                 } else {
                     cmdApp = "python3"; // Not shipped, assumed installed on Linux
-                    uploadOpts.unshift(espota); // Need to call Python3
+                    uploadOpts.unshift(espota + ".py"); // Need to call Python3
```

Comment:
Nice, I think you're almost there.  Thanks for digging into this!

Can you remove the `platform==darwin` bit at line 510?  There's no Mac executable (only espota.exe == win32, espota.py == all others) so it needs the .py put back on it and to call the python3 interpreter.  Just reading it the change looks good for Linux.